libtcmu_log.c

```diff
@@ -44,11 +50,27 @@ struct log_buf {
 	unsigned int head;
 	unsigned int tail;
 	char buf[LOG_ENTRYS][LOG_ENTRY_LEN];
+	darray(struct log_output) outputs;
 	pthread_t thread_id;
 };
 
+struct log_output {
+	log_output_func f;
+	log_close_func c;
+	int priority;
+	const char *name;
+	void *data;
+	tcmu_log_destination dest;
+};
+
 static int tcmu_log_level = TCMU_LOG_WARN;
 static struct log_buf *tcmu_log_initialize(void);
+static void close_syslog(void *);
+static void close_fd(void *);
+static int add_output_to_syslog(int priority, const char *ident);
+static int add_output_to_file(int priority, const char *ident);
+static int add_output_to_stdout(int priority);
+static const char *loglevel_string(int);
```

Comment:
Could you not forward declare everything?

- If the function is already defined before the user then definitely do not do it. For example close_syslog().
- If you can just move the definition of the functions around, then just do that. For example I think pretty much the reset of them like just put loglevel_string before output_to_fd.